neo/Consensus/ConsensusService.cs

```diff
@@ -74,31 +75,70 @@ private void ChangeTimer(TimeSpan delay)
 
         private void CheckExpectedView(byte view_number)
         {
+            Log($"CheckExpectedView: view_number={view_number} context.ViewNumber={context.ViewNumber} nv={context.ExpectedView[context.MyIndex]} state={context.State}");
+
             if (context.ViewNumber == view_number) return;
+            //if (context.State.HasFlag(ConsensusState.CommitSent))
+            //{
```

Comment:
We should lock this phase, but no replay again his signature. But locks include new possible problems, and recover state is not easy job

Reply:
Solved on last commit with OnRegeneration event